crates/biome_test_utils/src/lib.rs

```diff
@@ -249,12 +249,32 @@ fn get_js_like_paths_in_dir(dir: &Utf8Path) -> Vec<BiomePath> {
         .collect()
 }
 
+fn find_pnpm_workspace_catalog(fs: &dyn FileSystem) -> Option<Catalogs> {
+    let working_directory = fs.working_directory().unwrap_or_default();
+
+    for dir in working_directory.ancestors() {
+        let workspace_file = dir.join("pnpm-workspace.yaml");
+        if !fs.path_is_file(&workspace_file) {
+            continue;
+        }
```

Comment:
For tests, we need them to be as isolated as possible, and going up the file tree all the way searching for files breaks that.

We need to search for the `pnpm-workspace.yaml` that is adjacent to the file being tested. For example, for grabbing `package.json` for a test case, we have:

```
let package_json_file = input_file.with_extension("package.json");
```

Also, I think it needs to support the `yml` file extension too.

Reply:
> Also, I think it needs to support the yml file extension too.

I don't think so, pnpm only supports pnpm-workspace.yaml (exact match): https://pnpm.io/pnpm-workspace_yaml

Reply:
I was checking the internal implementation of pnpm, and it seemed to be fixed in pnpm-workspace.yaml.

https://github.com/pnpm/pnpm/blob/98a0410aa1e42784849c5def98341011144a3da4/packages/constants/src/index.ts#L12